graphql_compiler/query_pagination/pagination_planning.py

```diff
@@ -1,10 +1,39 @@
 # Copyright 2019-present Kensho Technologies, LLC.
 from typing import NamedTuple, Tuple
 
+from graphql import DocumentNode
+
 from ..ast_manipulation import get_only_query_definition, get_only_selection_from_ast
 from ..cost_estimation.helpers import is_uuid4_type
 from ..cost_estimation.int_value_conversion import field_supports_range_reasoning
 from ..exceptions import GraphQLError, GraphQLPaginationError
+from ..schema.schema_info import QueryPlanningSchemaInfo
+
+
+class PaginationWarning(object):
+    def __init__(self, message):
+        self.message = message
+
+
+class PaginationFieldNotSpecified(PaginationWarning):
+    def __init__(self, vertex_name):
+        super(PaginationFieldNotSpecified, self).__init__(
+            "Pagination field not specified for vertex {}".format(vertex_name)
+        )
+        self.vertex_name = vertex_name
+
+
+class NotEnoughQuantiles(PaginationWarning):
+    def __init__(self, vertex_name, field_name, current_resolution, desired_resolution):
+        super(NotEnoughQuantiles, self).__init__(
+            "Pagination would have been more successful if more quantiles were provided"
+            "For {}.{}. Currently there is {}, ideally there should be {}".format(
```

Comment:
How about `is/are` instead of `is`? I'm assuming the value is likely to be other than 1.